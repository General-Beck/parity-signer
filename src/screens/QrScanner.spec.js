import React from 'react';
import Adapter from "enzyme-adapter-react-16";
import { TouchableOpacity } from 'react-native';
import { shallow, configure, mount } from 'enzyme';
import QrScanner from './QrScanner';

import { JSDOM } from "jsdom";

const jsdom = new JSDOM("<!doctype html><html><body></body></html>");
const { window } = jsdom;

const copyProps = (src, target) => {
  const props = Object.getOwnPropertyNames(src)
    .filter(prop => typeof target[prop] === "undefined")
    .map((prop) => {
    	console.log(JSON.stringify(prop))
    	return Object.getOwnPropertyDescriptor(src, prop)
    });

  Object.defineProperties(target, props);
};

global.window = window;
global.document = window.document;
global.navigator = {
  userAgent: "node.js"
};
copyProps(window, global);

// Setup enzyme's react adapter
configure({ adapter: new Adapter() });

// Ignore React Web errors when using React Native
console.error = message => {
  return message;
};

describe('QrScanner Tests', () => {
  test('renders correctly', () => {
    const wrapper = shallow(<QrScanner />);
    expect(wrapper).toMatchSnapshot();
  });
});