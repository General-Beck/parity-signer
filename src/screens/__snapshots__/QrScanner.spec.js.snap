// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QrScanner Tests renders correctly 1`] = `ShallowWrapper {}`;
